lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -672,15 +672,23 @@ class PullbackCloner::Implementation final
   // Optional differentiation
   //--------------------------------------------------------------------------//
 
-  /// Given a `wrappedAdjoint` value of type `T.TangentVector`, creates an
-  /// `Optional<T>.TangentVector` value from it and adds it to the adjoint value
-  /// of `optionalValue`.
+  /// Given a `wrappedAdjoint` value of type `T.TangentVector` and `Optional<T>`
+  /// type, creates an `Optional<T>.TangentVector` buffer from it.
   ///
   /// `wrappedAdjoint` may be an object or address value, both cases are
   /// handled.
+  AllocStackInst *createOptionalAdjoint(SILBasicBlock *bb,
+                                        SILValue wrappedAdjoint,
+                                        SILType optionalTy);
+
+  /// Accumulate SILValueCategory::Address optional value from `wrappedAdjoint`.
   void accumulateAdjointForOptional(SILBasicBlock *bb, SILValue optionalValue,
                                     SILValue wrappedAdjoint);
 
+  /// Set SILValueCategory::Object optional value from `wrappedAdjoint`.
+  void setAdjointValueForOptional(SILBasicBlock *bb, SILValue optionalValue,
+                                  SILValue wrappedAdjoint);
```

Comment:
In AD code we use the term "adjoint value" to refer to adjoint objects, and "adjoint buffer" to refer to adjoint addresses, so I don't think there's a need to call out `SILValueCategory::Object` or `SILValueCategory::Address`.

What you could do is rename `accumulateAdjointForOptional` to make it clear at the call site, same for its arguments, for example:
```cpp
    void accumulateAdjointForOptionalBuffer(SILBasicBlock *bb, SILValue optionalBuffer,
                                            SILValue wrappedAdjointBuffer);

Reply:
Thank you! Done.